### PR TITLE
Added more events and commands to Sheet

### PR DESCRIPTION
**Pull request by haavamoa**
## Added / Fixed

- Sheet
    - Added `OnOpen` event
    - Added `OnClose` event
    - Added `OnBeforeOpen` event, command and command parameter
    - Added `OnBeforeClose` event, command and commandparameter
- ModalityLayout
    - Refactored to get more clean code

## Todo List

- [X] I have tested on an Android device.
- [X] I have tested on an iOS device.
- [ ] I have added unit tests

>*Please add pictures / Gifs if possible*
>*Todo List can be checked by putting a `X` inside the brackets*
>*Remember to update our `wiki` after this PR is merged*

## References
- #110

### CODE REVIEW

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs [hunk 1]
```diff
@@ -339,6 +428,31 @@ public void Hide()
...
+        public async Task BeforeRemoval()
+        {
+            if (m_modalityLayout == null) return;
+            if (m_sheetView == null) return;
+
+            var y = Alignment switch
+            {
+                AlignmentOptions.Bottom => m_modalityLayout.Height,
+                AlignmentOptions.Top => -m_modalityLayout.Height,
+                _ => throw new ArgumentOutOfRangeException()
```
Comment (Illedan) [inline src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: ArgumentOutOfRangeException?

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs [hunk 2]
```diff
@@ -339,6 +428,31 @@ public void Hide()
...
+            };
+
+            await m_sheetView.SheetFrame.TranslateTo(m_sheetView.SheetFrame.X, y);
```
Comment (Illedan) [inline src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: m_sheetView.SheetFrame.X ? Shouldn't this be Y?

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs [hunk 3]
```diff
@@ -339,6 +428,31 @@ public void Hide()
...
+        }
+
+        /// <inheritdoc />
+        public Task AfterRemoval()
+        {
+            OnCloseCommand?.Execute(OnCloseCommandParameter);
+            OnClose?.Invoke(this, new EventArgs());
```
Comment (Illedan) [inline src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: EventArgs.Empty

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs [hunk 4]
```diff
@@ -393,15 +507,14 @@ private async void ToggleSheetVisibility()
...
 
-            if (m_sheetView == null)
-            {
-                m_sheetView = new SheetView(this);
-            }
-
             if (IsOpen)
             {
+                OnBeforeOpenCommand?.Execute(OnBeforeOpenCommandParameter);
+                OnBeforeOpen?.Invoke(this, new EventArgs());
```
Comment (Illedan) [inline src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: EventArgs.Empty

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs [hunk 5]
```diff
@@ -393,15 +507,14 @@ private async void ToggleSheetVisibility()
...
+
+                m_sheetView = new SheetView(this);
```
Comment (Illedan) [inline src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: Keep local variable inside 1 scope.

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs [hunk 6]
```diff
@@ -419,31 +532,38 @@ private async void ToggleSheetVisibility()
...
-            else { 
-                var y = Alignment switch { 
-                        AlignmentOptions.Bottom => m_modalityLayout.Height,
-                        AlignmentOptions.Top => -m_modalityLayout.Height,
-                        _ => throw new ArgumentOutOfRangeException()
-                    };
-                m_modalityLayout.Hide(m_sheetView.SheetFrame, m_sheetView.SheetFrame.TranslateTo(m_sheetView.SheetFrame.X, y), () => OnCloseCommand?.Execute(OnCloseCommandParameter));
+            else
+            {
+                if (m_sheetView == null) return;
```
Comment (Illedan) [inline src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: Nullcheck? Didn't you just create the variable?

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs (hunk 1 above)
Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: Default from resharper when creating switches. I need a better default I guess :P

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs (hunk 2 above)
Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: No, first argument is `X` and second is `Y`. This is correct.

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs (hunk 5 above)
Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: What variablel? `m_sheet` is global and should be used for callback methods like `BeforeRemoval` etc.

File: src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs (hunk 6 above)
Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs]: No, I didn't. I create it in the `if` branch, and check for null and use it in the `else` branch,

**Pull request merged**